core/vm/contractstatefultests/allow_list_test.go

```diff
@@ -0,0 +1,284 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package contractstatefultests
+
+import (
+	"testing"
+
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/subnet-evm/core/rawdb"
+	"github.com/ava-labs/subnet-evm/core/state"
+	"github.com/ava-labs/subnet-evm/precompile"
+	"github.com/ava-labs/subnet-evm/precompile/allowlist"
+	"github.com/ava-labs/subnet-evm/vmerrs"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAllowListRun(t *testing.T) {
+	type test struct {
```

Comment:
Can we share the `test` type across all of the different precompile tests?

Reply:
done